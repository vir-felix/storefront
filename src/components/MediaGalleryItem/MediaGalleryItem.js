import React, { Component } from "react";
import PropTypes from "prop-types";
import ButtonBase from "material-ui/ButtonBase";
import withStyles from "material-ui/styles/withStyles";

const styles = () => ({
  root: {
    minWidth: 140
  },
  img: {
    width: "100%"
  }
});

/**
 * Product detail media gallery item
 * @class ProductDetailMediaGalleryItem
 */
@withStyles(styles)
class MediaGalleryItem extends Component {
  static propTypes = {
    /**
     * CSS class names
     */
    classes: PropTypes.object,

    /**
     * Product media
     */
    media: PropTypes.object,

    /**
     * Click callback
     * @example (event, media) => {}
     */
    onClick: PropTypes.func
  }

  static defaultProps = {
    onClick: () => {}
  }

  /**
   * Click handler for ButtonBase
   * @param {SyntheticEvent} event Event
   * @returns {undefined}
   */
  handleClick = (event) => {
    this.props.onClick(event, this.props.media);
  }

  render() {
    const { classes, media } = this.props;

    // If all props are undefined then skip rendering component
    if (!media) return null;

    return (
      <ButtonBase className={classes.root} onClick={this.handleClick}>
        <img
          className={classes.img}
          src={media.url}
          alt=""
        />
      </ButtonBase>
    );
  }
}

export default MediaGalleryItem;
