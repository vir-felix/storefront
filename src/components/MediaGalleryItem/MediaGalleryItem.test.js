import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "mobx-react";
import MediaGalleryItem from "./MediaGalleryItem";

const uiStore = {
  appConfig: {
    publicRuntimeConfig: {
      placeholderImageUrls: {
        productGrid: ""
      }
    }
  }
};

const media = {
  toGrid: 1,
  priority: 0,
  productId: "cmVhY3Rpb24vcHJvZHVjdDpCQ1RNWjZIVHhGU3BwSkVTaw==",
  variantId: "cmVhY3Rpb24vcHJvZHVjdDo2cWlxUHdCa2VKZHRkUWM0Rw==",
  URLs: {
    large: "/resources/placeholder.gif",
    medium: "/resources/placeholder.gif",
    original: "/resources/placeholder.gif",
    small: "/resources/placeholder.gif",
    thumbnail: "/resources/placeholder.gif"
  }
};

test("basic snapshot", () => {
  const component = renderer.create(<Provider uiStore={uiStore}>
    <MediaGalleryItem media={media} />
  </Provider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
