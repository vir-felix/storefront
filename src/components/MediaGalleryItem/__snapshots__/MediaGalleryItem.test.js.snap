// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<button
  className="MuiButtonBase-root-3 inject-MediaGalleryItem-with-uiStore-root-1"
  disabled={undefined}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <img
    alt=""
    className="inject-MediaGalleryItem-with-uiStore-img-2"
    src="/assets/files/Media/P3MkhDWadHoDNsFPE/small/I0DhYYC.png"
  />
  <span
    className="MuiTouchRipple-root-6"
  />
</button>
`;
